packages/SwingSet/src/controller/initializeSwingset.js

```diff
@@ -240,6 +240,29 @@ export function swingsetIsInitialized(hostStorage) {
   return !!hostStorage.kvStore.get('initialized');
 }
 
+/**
+ * @param {Record<string, any>} obj
+ * @param {string} [first]
+ * @returns {Record<string, any>}
+ */
+function sortObjectProperties(obj, first) {
```

Comment:
It'd be even nicer if the order could be `${bootstrap}, vat-admin, comms, vattp, timer, ...uservats`. What about (note: untested):

```js
function sortObjectProperties(obj, firsts) {
  const sorted = [...firsts, Object.keys(obj).sort()]; // may have duplicates
  const result = {};
  for (const prop of sorted) {
    // don't add duplicate && only add if actually present
    if (!Object.hasOwn(result, prop) && Object.hasOwn(obj, prop)) {
      result[prop] = obj[prop];
    };
  }
  return result;
}
```

where the second test allows us to later say:
```js
config.vats = sortObjectProperties(config.vats, [config.bootstrap, 'vat-admin', 'comms', 'vattp', 'timer']);
```

without knowing whether those five vat names are actually present in `config.vats` or not.

Reply:
I did something like that, but for future reference, XS does not implement `Object.hasOwn` yet.